edb/server/pgcon/pgcon.pyx

```diff
@@ -90,6 +90,14 @@ cdef bytes INIT_CON_SCRIPT = None
 
 def _build_init_con_script() -> bytes:
     return (f'''
+        SELECT CASE WHEN pg_is_in_recovery() THEN
```

Comment:
Perhaps make this conditional on PostgreSQL<=13 to avoid paying the price of the query on modern versions.